### ISSUE TITLE
abort on NetCDF file name containing "="

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#784]
**Reported by Ansley B Manke on 15 May 2007 18:23 UTC**
(reported by Andrew Wittenberg)

yes? sp touch a=b.nc
yes? use "a=b.nc"
terminate called after throwing an instance of 'Error'
Abort

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/1512

**Comment by karlmsmith** [comment]
**Comment by Ansley.B.Manke@noaa.gov on 17 May 2007 17:51 UTC**
Just additional info -- this crash is inside the OPeNDAP netcdf library.  An
ascii file can have an = sign in the name

yes? list/clobber/i=1:5/file=five.dat/format=(f5.1)/nohead/norowhead i
 LISTing to file five.dat
yes? sp cp five.dat f=1.dat
yes? file/var=v1 f=1.dat
yes? list v1
             VARIABLE : V1
             FILENAME : f=1.dat
             SUBSET   : 5 points (X)
 1   / 1:  1.000
 2   / 2:  2.000
 3   / 3:  3.000
 4   / 4:  4.000
 5   / 5:  5.000


For a netcdf file, here is the debugger Totalview stack trace; cd_open_dset is
the Ferret routine which calls the netcdf  nf_open.

     abort,                FP=bfff48f8
     __cxa_call_unexpected, FP=bfff4918
     _ZSt9terminatev,      FP=bfff4928
     __cxa_throw,          FP=bfff4948
ClientParams::ClientParams, FP=bfff4b38
nc__open_mp,          FP=bfff4bc8
nc__open,             FP=bfff4bf8
nc_open,              FP=bfff4c18
nf_open__,            FP=bfff4c48
cd_open_dset_,        FP=bfff5098

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 18 Jan 2009 18:06 UTC**
Bumping up the priority on this, as we've discovered many legacy NetCDF files that have names containing "=".

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 8 Sep 2009 17:37 UTC**
This is NOT a Ferret error. The crash occurs inside the OPeNDAP library. Ferret passes the file name to the nc_open routine of the OPeNDAP library and that routine is unable to open the file.

With V6.2 of Ferret 

```
yes? use "a=b.nc"
terminate called after throwing an instance of 'libdap::Error'
Abort
```

The same error occurs if we try "ncdump" on the file.

However I am experimenting with the latest netcdf-4 libraries. These have other problems for Ferret just now, and I am working with the developers to resolve those. Opening a file whose name contains = does seem to work:

```
/home/porter/ansley/build/x86_gfortran/FERRET/fer/ferret_c
        NOAA/PMEL TMAP
        FERRET v6.205 (alpha)
        Linux(g77) 2.6.9-22. - 09/02/09
         8-Sep-09 10:34     

yes? use a=b.nc
yes? sh dat
     currently SET data sets:
    1> ./a=b.nc  (default)
 name     title                             I         J         K         L
 AA       {1,3,6,7}                        1:4       ...       ...       ...
 
yes? list aa
             VARIABLE : {1,3,6,7}
             FILENAME : a=b.nc
             SUBSET   : 4 points (X)
 1   / 1:  1.000
 2   / 2:  3.000
 3   / 3:  6.000
 4   / 4:  7.000

```

**Comment by karlmsmith** [comment]
**Comment by andrew.wittenberg on 2 Feb 2010 22:34 UTC**
This seems to be fixed in v6.43beta, with the new NetCDF library.
